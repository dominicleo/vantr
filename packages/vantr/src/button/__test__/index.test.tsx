import * as React from 'react';
import { fireEvent, render } from '@testing-library/react';
import Button from '../';

describe('Button', () => {
  it('snapshot', () => {
    const component = render(<Button />);
    expect(component.asFragment()).toMatchSnapshot();
  });

  it('onPress', () => {
    const onPress = jest.fn();
    const component = render(<Button onPress={onPress}>button</Button>);
    const element = component.getByText('button');

    fireEvent.click(element);

    expect(onPress).toHaveBeenCalled();
  });
});
