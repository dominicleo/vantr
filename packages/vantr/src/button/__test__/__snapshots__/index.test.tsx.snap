// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<DocumentFragment>
  <button
    classes="vanr-button"
  />
</DocumentFragment>
`;
